.github/workflows/go-build.yaml

```
@@ -75,14 +80,16 @@ jobs:
+      - name: docker-auth
         if: ${{ inputs.DOCKER_MANIFEST }}
         run: |
           echo "GCP_PROJECT_ID=$($MAKE signoz-gcp-project-id)" >> $GITHUB_ENV
           echo "GCP_WORKLOAD_IDENTITY_PROVIDER=$($MAKE signoz-gcp-workload-identity-provider)" >> $GITHUB_ENV
-          echo "GCP_SERVICE_ACCOUNT=$($MAKE signoz-gcp-service-account)" >> $GITHUB_ENV
+          echo "GCP_SERVICE_ACCOUNT=$($MAKE signoz-gcp-service-account)" >> $GITHUB_ENV"
+          echo "HUB_USERNAME=${{ secrets.DOCKERHUB_USERNAME }}" >> $GITHUB_ENV
+          echo "HUB_TOKEN=${{ secrets.DOCKERHUB_TOKEN }}" >> $GITHUB_ENV
       - name: gcp-auth
-        if: ${{ inputs.DOCKER_MANIFEST }}
+        if: ${{ inputs.DOCKER_MANIFEST }} && contains(inputs.DOCKER_PROVIDERS, 'gcp')
```

actionlint (1.7.4)

if: condition "${{ inputs.DOCKER_MANIFEST }} && contains(inputs.DOCKER_PROVIDERS, 'gcp')" is always evaluated to true because extra characters are around ${{ }}

(if-cond)